pandas/tests/indexes/test_index_new.py

```diff
@@ -122,6 +126,80 @@ def test_constructor_mixed_nat_objs_infers_object(self, swap_objs):
         tm.assert_index_equal(Index(np.array(data, dtype=object)), expected)
 
 
+class TestDtypeEnforced:
```

Comment:
slightly OT: wha tis the purpose of this specific test file, should this not be in base_class/ or maybe test_construction.py?

Reply:
this file is aimed at `Index.__new__`, in particular checking it returns the expected subclass

Reply:
is the file docstring unclear?

Reply:
yes, but the *name* of the file is not super clear as 'new' indicates not ```___new__```, would probably rename to test_construction.py

Reply:
ok, ill look at renaming in upcoming tests pass